force-app/main/default/lwc/hero/__tests__/hero.test.js

```diff
@@ -0,0 +1,186 @@
+        return Promise.resolve().then(() => {
+            // Select elements for validation
+            const imageEl = element.shadowRoot.querySelector('img');
+            expect(imageEl).not.toBeNull();
+            // Verify that the URL returned matches, in the context of the test, the domain will render as http://localhost/
+            expect(imageEl.src).toBe(
+                'http://localhost/' + IMAGE_URL + mockResourceUrl
+            );
+        });
+    });
+
+    it('displays an video in the hero when the type is video', () => {
+        const element = createElement('c-hero', {
+            is: Hero
+        });
+        element.title = mockTitle;
+        element.slogan = mockSlogan;
+        element.buttonText = mockButtonText;
+        element.heroDetailsPosition = mockHeroDetailsPositionLEFT;
+        element.resourceUrl = mockResourceUrl;
+        element.imgOrVideo = mockImgOrVideoVIDEO;
+        element.internalResource = mockInternalResource;
+        element.overlay = mockOverlay;
+        element.opacity = mockOpacity;
+        element.buttonClickProductOrFamilyName = mockButtonClickProductOrFamilyName;
+        document.body.appendChild(element);
+
+        // Return a promise to wait for any asynchronous DOM updates. Jest
+        // will automatically wait for the Promise chain to complete before
+        // ending the test and fail the test if the promise rejects.
+        return Promise.resolve().then(() => {
+            // Select elements for validation
+            const sourceEl = element.shadowRoot.querySelector('source');
+            expect(sourceEl).not.toBeNull();
+            // Verify that the URL returned matches, in the context of the test, the domain will render as http://localhost/
+            expect(sourceEl.src).toBe('http://localhost/' + mockResourceUrl);
+        });
+    });
+
+    it('displays an overlay', () => {
+        const element = createElement('c-hero', {
+            is: Hero
+        });
+        element.title = mockTitle;
+        element.slogan = mockSlogan;
+        element.buttonText = mockButtonText;
+        element.heroDetailsPosition = mockHeroDetailsPositionLEFT;
+        element.resourceUrl = mockResourceUrl;
+        element.imgOrVideo = mockImgOrVideoVIDEO;
+        element.internalResource = mockInternalResource;
+        element.overlay = mockOverlay;
+        element.opacity = mockOpacity;
+        element.buttonClickProductOrFamilyName = mockButtonClickProductOrFamilyName;
+        document.body.appendChild(element);
+
+        // Return a promise to wait for any asynchronous DOM updates. Jest
+        // will automatically wait for the Promise chain to complete before
+        // ending the test and fail the test if the promise rejects.
+        return Promise.resolve().then(() => {
+            // Select elements for validation
+            const divEl = element.shadowRoot.querySelector('div');
+            expect(divEl).not.toBeNull();
```

Comment:
also check for `overlay.style.opacity` value

Reply:
I'm looking for a good way to test the style, but not finding one. Do we have any examples?

Reply:
No particular example at hand but it should be pretty simple to test, you just need to extend that test:
1. Set opacity to `5` when creating the component
1. Add the following check at the end
```js
expect(divEl.style.opacity).toBe(0.5)
```

Reply:
Great, that's sorted!